packages/ra-core/src/controller/useEditController.ts

```diff
@@ -0,0 +1,132 @@
+import { useEffect, useCallback } from 'react';
+// @ts-ignore
+import { useDispatch } from 'react-redux';
+import { reset as resetForm } from 'redux-form';
+import inflection from 'inflection';
+
+import useVersion from './useVersion';
+import { useCheckMinimumRequiredProps } from './checkMinimumRequiredProps';
+import { REDUX_FORM_NAME } from '../form';
+import { Record, Identifier } from '../types';
+import { RedirectionSideEffect } from '../sideEffect';
+import { useGetOne, useUpdate } from '../fetch';
+import { useTranslate } from '../i18n';
+
+export interface EditProps {
+    basePath: string;
+    hasCreate?: boolean;
+    hasEdit?: boolean;
+    hasShow?: boolean;
+    hasList?: boolean;
+    id: Identifier;
+    resource: string;
+    undoable?: boolean;
+    [key: string]: any;
+}
+
+export interface EditControllerProps {
+    isLoading: boolean;
+    isSaving: boolean;
+    defaultTitle: string;
+    save: (data: Record, redirect?: RedirectionSideEffect) => void;
+    resource: string;
+    basePath: string;
+    record?: Record;
+    version: number;
+}
+
+/**
+ * Prepare data for the Edit view
+ *
+ * @param {Object} props The props passed to the Edit component.
+ *
+ * @return {Object} controllerProps Fetched data and callbacks for the Edit view
+ *
+ * @example
+ *
+ * import { useEditController } from 'react-admin';
+ * import EditView from './EditView';
+ *
+ * const MyEdit = props => {
+ *     const controllerProps = useEditController(props);
+ *     return <EditView {...controllerProps} {...props} />;
+ * }
+ */
+const useEditController = (props: EditProps): EditControllerProps => {
+    useCheckMinimumRequiredProps('Edit', ['basePath', 'resource'], props);
+    const { basePath, id, resource, undoable = true } = props;
+    const translate = useTranslate();
+    const dispatch = useDispatch();
+    const version = useVersion();
+    const { data: record, loading } = useGetOne(resource, id, {
+        basePath,
+        version, // used to force reload
+        onFailure: {
+            notification: {
+                body: 'ra.notification.item_doesnt_exist',
+                level: 'warning',
+            },
+            redirectTo: 'list',
+            refresh: true,
+        },
+    });
+
+    useEffect(() => {
+        dispatch(resetForm(REDUX_FORM_NAME));
+    }, [resource, id, version]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
You should pass the missing deps

Reply:
No, because the previous code was:

```jsx
    componentWillReceiveProps(nextProps: Props & EnhancedProps) {
        if (
            this.props.id !== nextProps.id ||
            nextProps.version !== this.props.version
        ) {
            this.props.resetForm(REDUX_FORM_NAME);
            this.updateData(nextProps.resource, nextProps.id);
        }
    }
```

Reply:
I fail to see how it relates to the new hook version and to what hooks actually requires

Reply:
Just read this from an article by Dan:
> You may omit dispatch, setState, and useRef container values from the deps because React guarantees them to be static. But it also doesn’t hurt to specify them.)

https://overreacted.io/a-complete-guide-to-useeffect/

Maybe it's true for react-redux `dispatch` as well